tests/InferTypesTest.elm

```diff
@@ -57,6 +57,20 @@ typeInference =
                   , Canonical.List [ Canonical.List [ Canonical.Literal (Int 1) ], Canonical.List [ Canonical.Literal (Bool False) ] ]
                   , Err (Error.TypeMismatch Type.Int Type.Bool)
                   )
+                , ( "List Concat"
+                  , Canonical.ListConcat (Canonical.List []) (Canonical.List [ Canonical.Literal (Bool True) ])
+                  , Ok ( Typed.ListConcat ( Typed.List [], Type.List <| Type.Bool ) ( Typed.List [ ( Typed.Literal (Bool True), Type.Bool ) ], Type.List <| Type.Bool ), Type.List <| Type.Bool )
```

Comment:
extraneous `<|` in a few places (ie. `Type.List <| Type.Bool` can be `Type.List Type.Bool`)